tests/api/db/test_sqldb.py

```diff
@@ -108,83 +120,96 @@ def test_run_iter0(db: SQLDB, db_session: Session):
+    )
 
     result = db.read_artifact(db_session, k1, "tag1", iter=1, project=prj)
-    assert result["tag"] == "tag1"
+    assert result["metadata"]["tag"] == "tag1"
     result = db.read_artifact(db_session, k1, "tag2", iter=2, project=prj)
-    assert result["tag"] == "tag2"
+    assert result["metadata"]["tag"] == "tag2"
     result = db.read_artifact(db_session, k1, iter=1, project=prj)
     # When doing get without a tag, the returned object must not contain a tag.
     assert "tag" not in result
-    # read_artifact supports a case where the tag is actually the uid.
-    result = db.read_artifact(db_session, k1, tag="u2", iter=2, project=prj)
-    assert "tag" not in result
 
     result = db.read_artifact(db_session, k1, "tag2", iter=2, project=prj)
-    assert result["tag"] == "tag2"
+    assert result["metadata"]["tag"] == "tag2"
 
     result = db.list_artifacts(db_session, k1, project=prj, tag="*")
-    # TODO: this actually expected to be 3, but this will be fixed in another PR once we restructure the Artifacts table
-    #   and change the way tag_artifacts works ( currently it is unable to untag tag from artifacts which were generated
-    #   in hyper params runs)
-    assert len(result) == 4
+    assert len(result) == 3
     for artifact in result:
         assert (
-            (artifact["a"] == 1 and artifact["tag"] == "tag1")
-            or (artifact["a"] == 2 and artifact["tag"] == "tag2")
-            or (artifact["a"] in (1, 2) and artifact["tag"] == "latest")
+            (artifact["a"] == 1 and artifact["metadata"]["tag"] == "tag1")
+            or (artifact["a"] == 2 and artifact["metadata"]["tag"] == "tag2")
+            or (artifact["a"] in (1, 2) and artifact["metadata"]["tag"] == "latest")
         )
 
     # To be used later, after adding tags
     full_results = result
 
     result = db.list_artifacts(db_session, k1, tag="tag1", project=prj)
-    assert len(result) == 1 and result[0]["tag"] == "tag1" and result[0]["a"] == 1
+    assert (
+        len(result) == 1
+        and result[0]["metadata"]["tag"] == "tag1"
+        and result[0]["a"] == 1
+    )
     result = db.list_artifacts(db_session, k1, tag="tag2", project=prj)
-    assert len(result) == 1 and result[0]["tag"] == "tag2" and result[0]["a"] == 2
+    assert (
+        len(result) == 1
+        and result[0]["metadata"]["tag"] == "tag2"
+        and result[0]["a"] == 2
+    )
 
     # Add another tag to all objects (there are 2 at this point)
     expected_results = ArtifactList()
     for artifact in full_results:
         expected_results.append(artifact)
-        artifact_with_new_tag = artifact.copy()
-        artifact_with_new_tag["tag"] = "new-tag"
+        if artifact["metadata"]["tag"] == "latest":
+            # We don't want to add a new tag to the "latest" object (it's the same object as the one with tag "tag2")
+            continue
+        artifact_with_new_tag = copy.deepcopy(artifact)
```

Comment:
cant use artifact.copy() like before?

Reply:
`artifact` is a dict, so it doesn't have the `copy` method.